src/pyscheduling/PMSP/PM_methods.py

```diff
@@ -148,92 +149,88 @@ def grasp(instance: pm.ParallelInstance, p: float = 0.5, r: float = 0.5, n_itera
 
 class Metaheuristics():
 
-    @staticmethod
-    def lahc(instance: pm.ParallelInstance, **kwargs):
-        """ Returns the solution using the LAHC algorithm
-
-        Args:
-            instance (ParallelInstance): Instance object to solve
-            Lfa (int, optional): Size of the candidates list. Defaults to 25.
-            n_iterations (int, optional): Number of iterations of LAHC. Defaults to 300.
-            Non_improv (int, optional): LAHC stops when the number of iterations without improvement is achieved. Defaults to 50.
-            LS (bool, optional): Flag to apply local search at each iteration or not. Defaults to True.
-            time_limit_factor: Fixes a time limit as follows: n*m*time_limit_factor if specified, else n_iterations is taken Defaults to None
-            init_sol_method: The method used to get the initial solution. Defaults to "constructive"
-            seed (int, optional): Seed for the random operators to make the algo deterministic
-
-        Returns:
-            Problem.SolveResult: the solver result of the execution of the metaheuristic
-        """
-        # Extracting parameters
-        time_limit_factor = kwargs.get("time_limit_factor", None)
-        init_sol_method = kwargs.get("init_sol_method", instance.init_sol_method())
-        Lfa = kwargs.get("Lfa", 30)
-        n_iterations = kwargs.get("n_iterations", 500000)
-        Non_improv = kwargs.get("Non_improv", 50000)
-        LS = kwargs.get("LS", True)
-        seed = kwargs.get("seed", None)
+    class LAHC(Solver):
```

Comment:
In general I would prefer if we keep restructuring and refactoring in its own PR. But lets keep this change as it is going in the right direction.

When we discuss restructuring later, I will that instead of having to define inner classes for every metaheuristic, we define a metaheuristics package and put the classes there. Probably the same for heuristics

Reply:
I wanted to test the listeners and needed to inherit from Solver at least one metaheuristic, I'm waiting for you and Mourad to move the solving methods to the abstract level before adding listeners everywhere. For now it's only done for LAHC in PMSP for testing purpose.